lib/sql.js

```diff
@@ -603,11 +603,25 @@ SQLConnector.prototype.updateAttributes = function(model, id, data, options, cb)
  * @private
  */
 SQLConnector.prototype.replaceById = function(model, id, data, options, cb) {
+  var self = this;
   if (!isIdValuePresent(id, cb)) return;
   var where = this._buildWhereObjById(model, id, data);
-  this._replace(model, where, data, options, cb);
+  this.find(model, where.id, options, function(err, foundInstance) {
```

Comment:
@bajtos @loay 

This is not the right solution as I mentioned in my previous comments in https://github.com/strongloop/loopback-datasource-juggler/pull/1210; please see my previous comments: https://github.com/strongloop/loopback-datasource-juggler/pull/1210#pullrequestreview-16251738 and https://github.com/strongloop/loopback-datasource-juggler/pull/1210#pullrequestreview-16039656 ... It definitely does not make sense to call another query (ie. find) to see the existence of the instance before replacing; as I mentioned in the cited comments when you call replace method, the db will return the number of affected rows/instances and if the number is 0 it means the PK/id of provided instance was not valid and there you need to return 404 with appropriate message (exactly what @bajtos explained in the previous comment as well ☝️ ). I'm on vacation, please work with @jannyHou or @bajtos and they can help review the code. I believe @bajtos has the same opinion as I mentioned in this comment and my previous comments: https://github.com/strongloop/loopback-datasource-juggler/pull/1210#pullrequestreview-16039656 and https://github.com/strongloop/loopback-datasource-juggler/pull/1210#pullrequestreview-16251738